flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetAggFunction.scala

```diff
@@ -97,12 +94,7 @@ class DataSetAggFunction(
         }
 
         // set agg results to output
-        i = 0
-        while (i < aggOutMapping.length) {
-          val (out, in) = aggOutMapping(i)
-          output.setField(out, aggregates(in).getValue(accumulators(in)))
-          i += 1
-        }
+        function.setAggregationResults(accumulators, output)
 
         // set grouping set flags to output
         if (intermediateGKeys.isDefined) {
```

Comment:
I think this should eventually be integrated with `setForwardFields()` as well. 
For now, we might leave it as it is.